intfconfd/intfconf.cpp

```diff
@@ -0,0 +1,120 @@
+#include <string.h>
+#include "logger.h"
+#include "dbconnector.h"
+#include "producerstatetable.h"
+#include "tokenize.h"
+#include "ipprefix.h"
+#include "intfconf.h"
+#include "exec.h"
+
+using namespace std;
+using namespace swss;
+
+#define VLAN_PREFIX         "Vlan"
+#define LAG_PREFIX          "PortChannel"
+
+IntfConf::IntfConf(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, vector<string> tableNames) :
+        CfgOrch(cfgDb, tableNames),
+        m_cfgIntfTable(cfgDb, CFG_INTF_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_cfgVlanIntfTable(cfgDb, CFG_VLAN_INTF_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_statePortTable(stateDb, STATE_PORT_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_stateVlanTable(stateDb, STATE_VLAN_TABLE_NAME, CONFIGDB_TABLE_NAME_SEPARATOR),
+        m_appIntfTableProducer(appDb, APP_INTF_TABLE_NAME)
+{
+
+}
+
+void IntfConf::syncCfgDB()
+{
+    CfgOrch::syncCfgDB(CFG_INTF_TABLE_NAME, m_cfgIntfTable);
+    CfgOrch::syncCfgDB(CFG_VLAN_INTF_TABLE_NAME, m_cfgVlanIntfTable);
+}
+
+bool IntfConf::setIntfIp(string &alias, string &opCmd, string &ipPrefixStr)
+{
+    string cmd, res;
+
+    cmd = "ip address " + opCmd + " ";
+    cmd += ipPrefixStr + " dev " + alias;
+    swss::exec(cmd, res);
```

Comment:
>exec [](start = 10, length = 4)

check the return value?

Reply:
Error has been logged in exec if there is any. In this case, error is not really expected.
That is common question for many functions in sonic, end 2 end feedback loop is missing.

Reply:
I don't agree. Locally if there is something error happens, you should check the return code all the levels up the stack, and quit as soon as possible.

Reply:
fine.  But I don't see the same criteria applied in orchagent, syncd, those python code and etc.
We could start the endeavor from this.